src/Compilers/Core/Portable/CommandLine/SarifV2ErrorLogger.cs

```diff
@@ -85,6 +114,14 @@ public override void LogDiagnostic(Diagnostic diagnostic, SuppressionInfo? suppr
             _writer.WriteObjectEnd(); // result
         }
 
+        public override void AddAnalyzerDescriptors(ImmutableArray<(DiagnosticDescriptor Descriptor, bool HasAnyExternalSuppression)> descriptors)
+        {
+            foreach (var (descriptor, hasAnyExternalSuppression) in descriptors.OrderBy(d => d.Descriptor.Id))
```

Comment:
>.OrderBy(d => d.Descriptor.Id) [](http://example.com/codeflow?start=79&length=30)

Is `OrderBy()` necessary, or will `_descriptors.Add()` store the items in a dictionary that ignores the order?

Reply:
`_descriptors.Add()` incrementally assigns an index to each added entry, and if we don't invoke `OrderBy` here, the indices will not be sorted by ID, and the rule entries in the error log will also be in a non-deterministic order.